### ISSUE TITLE
led attributes might be out of sync with the kernel

### ISSUE THREAD

**Issue by ensonic** [rhempel/ev3dev-lang-python#126]
On a ev3:
```
ls -al /sys/class/leds/ev3:left:red:ev3dev/
total 0
drwxr-xr-x 3 root root      0 Jan 21 09:43 .
drwxr-xr-x 6 root root      0 Jan  1  2000 ..
-rw-rw-r-- 1 root ev3dev 4096 Jan 21 09:08 brightness
lrwxrwxrwx 1 root root      0 Jan  1  2000 device -> ../../../leds_pwm
-r--r--r-- 1 root ev3dev 4096 Jan  1  2000 max_brightness
drwxr-xr-x 2 root root      0 Jan  1  2000 power
lrwxrwxrwx 1 root root      0 Jan  1  2000 subsystem -> ../../../../../class/leds
-rw-rw-r-- 1 root ev3dev 4096 Jan 21 09:08 trigger
-rw-rw-r-- 1 root ev3dev 4096 Jan  1  2000 uevent
```

but on the python api:
```
>>> dir(ev3dev.Led)
['DEVICE_ROOT_PATH', 'SYSTEM_CLASS_NAME', 'SYSTEM_DEVICE_NAME_CONVENTION', '_DEVICE_INDEX', '__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_get_attribute', '_matches', '_set_attribute', 'brightness', 'brightness_pct', 'delay_off', 'delay_on', 'device_index', 'get_attr_from_set', 'get_attr_int', 'get_attr_line', 'get_attr_set', 'get_attr_string', 'max_brightness', 'set_attr_int', 'set_attr_string', 'trigger', 'triggers']
```

The trigger attribute can be set, but if I e.g. try to do:
```
leds.set(ev3dev.Leds.RIGHT, trigger='timer', delay_on=500, delay_off=500)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/ev3dev/ev3.py", line 89, in set
    setattr(led, k, kwargs[k])
  File "/usr/lib/python2.7/dist-packages/ev3dev/core.py", line 1763, in delay_off
    self.set_attr_int('delay_off', value)
  File "/usr/lib/python2.7/dist-packages/ev3dev/core.py", line 210, in set_attr_int
    self._set_attribute(attribute, '{0:d}'.format(int(value)))
  File "/usr/lib/python2.7/dist-packages/ev3dev/core.py", line 204, in _set_attribute
    self._attribute_cache.write(abspath(self._path + '/' + attribute), value)
  File "/usr/lib/python2.7/dist-packages/ev3dev/core.py", line 91, in write
    f.write(value.encode())
IOError: File not open for writing
```

And indeed there are no 'delay_on/off' sys-fs entries. Maybe I am missing somethoing, if so we need more examples :)

**Comment by ddemidov** [comment]
I think the `delay_{on,off}` attributes are special in the sense they only appear after you set the trigger to `timer`. The led interface is standard linux one. It has some known issues, and I think at some point @dlech recommended to stay away from triggers and do the led flashing in user code, but I can not find the exact comment.

Will your code work if you split it into
```py
leds.set(ev3dev.Leds.RIGHT, trigger='timer')
leds.set(ev3dev.Leds.RIGHT, delay_on=500, delay_off=500)
```
?

**Comment by ensonic** [comment]
Yep, that work. Since we have that example in the docs:
  1. we either expand it
  1. drop it
  1. add a note that using it is not recommended

**Comment by ddemidov** [comment]
I can not find the trigger example in the docs. Can you point me to it?

**Comment by ensonic** [comment]
https://github.com/rhempel/ev3dev-lang-python/blob/master/ev3dev/ev3.py#L79-L86

**Comment by ddemidov** [comment]
Looks like sphinx autodoc does not show docstrings for property setters. I guess its better to merge getter/setter docstrings then.

**Comment by ddemidov** [comment]
It looks like sphinx fails to document decorated methods. See the note here: http://www.sphinx-doc.org/en/stable/ext/autodoc.html#directive-autoattribute

**Comment by dlech** [comment]
It is mostly the `timer` trigger that should be avoided because it adds attributes later and there is a race condition between the udev rules setting permissions on the new attributes and the program trying to write to the attributes.

**Comment by ensonic** [comment]
I think now this is completely broken:
``` python
from ev3dev import auto as ev3dev
led = ev3dev.Leds
led.set([led.green_left, led.green_right], trigger='timer')
```
yields:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/ev3dev/ev3.py", line 90, in set
    setattr(led, k, kwargs[k])
  File "/usr/lib/python3/dist-packages/ev3dev/core.py", line 2027, in trigger
    self._trigger = self.set_attr_from_set(self._trigger, 'trigger', value)
AttributeError: 'Led' object has no attribute 'set_attr_from_set'
```

Seems that I really have to add a thread to do led blinking. Double-meh.

**Comment by ddemidov** [comment]
Sorry for that, 9b8deb4 fixed that for me.

**Comment by ensonic** [comment]
I hope the fix is part of 0.7, right?

**Comment by ddemidov** [comment]
It is.
